hardly/tasks.py

```diff
@@ -0,0 +1,107 @@
+# Copyright Contributors to the Packit project.
+# SPDX-License-Identifier: MIT
+
+import logging
+from os import getenv
+from typing import List
+
+from celery import Task
+
+from hardly.handlers.abstract import TaskName
+from hardly.handlers.distgit import DistGitPRHandler
+from hardly.jobs import StreamJobs
+from packit_service.celerizer import celery_app
+from packit_service.constants import DEFAULT_RETRY_LIMIT, DEFAULT_RETRY_BACKOFF
+from packit_service.utils import load_job_config, load_package_config
+from packit_service.worker.handlers import (
+    CoprBuildEndHandler,
+    CoprBuildStartHandler,
+    CoprBuildHandler,
+)
+from packit_service.worker.result import TaskResults
+
+logger = logging.getLogger(__name__)
+
+# debug logs of these are super-duper verbose
+logging.getLogger("requests").setLevel(logging.WARNING)
+logging.getLogger("urllib3").setLevel(logging.WARNING)
+logging.getLogger("github").setLevel(logging.WARNING)
+logging.getLogger("kubernetes").setLevel(logging.WARNING)
+logging.getLogger("botocore").setLevel(logging.WARNING)
+# info is just enough
+logging.getLogger("ogr").setLevel(logging.INFO)
+# easier debugging
+logging.getLogger("packit").setLevel(logging.DEBUG)
+logging.getLogger("sandcastle").setLevel(logging.DEBUG)
+
+
+# Don't import this (or anything) from p_s.worker.tasks,
+# it would create the task from their process_message()
+class HandlerTaskWithRetry(Task):
+    autoretry_for = (Exception,)
+    retry_kwargs = {
+        "max_retries": int(getenv("CELERY_RETRY_LIMIT", DEFAULT_RETRY_LIMIT))
+    }
+    retry_backoff = int(getenv("CELERY_RETRY_BACKOFF", DEFAULT_RETRY_BACKOFF))
+
+
+@celery_app.task(name="task.steve_jobs.process_message", bind=True)
+def process_message(
+    self, event: dict, topic: str = None, source: str = None
+) -> List[TaskResults]:
+    """
+    Base celery task for processing messages.
+
+    :param event: event data
+    :param topic: event topic
+    :param source: event source
+    :return: dictionary containing task results
+    """
+    return StreamJobs().process_message(event=event, topic=topic, source=source)
```

Comment:
What about using a different name than `process_message` to not collide and be more explicit when calling this.

Or, is this for some reason?

Reply:
Not sure I understand. We don't collide, we override it. But I renamed it (to `hardly_process`) if that's better for you.

Reply:
I meant the `process_message` in `tasks.py` and the task name and was not sure if this is the same for purpose. (And as you wrote above, we do that to override it. That answered my question...;-)